src/laboratory/static/laboratory/js/laboratory.js

```diff
@@ -1,3 +1,154 @@
+function convertFormToJSON(form, prefix="") {
+  const re = new RegExp("^"+prefix);
+  return form
+    .serializeArray()
+    .reduce(function (json, { name, value }) {
+      json[name.replace(re, "")] = value;
+      return json;
+    }, {});
+}
+
+function convertToStringJson(form, prefix="", extras={}){
+    var formjson =convertFormToJSON(form, prefix=prefix);
+    formjson=Object.assign({}, formjson, extras)
+    return JSON.stringify(formjson);
+}
+
+
+function load_errors(error_list, obj){
+    ul_obj = "<ul class='errorlist shelf_form_errors d-flex justify-content-center'>";
+    error_list.forEach((item)=>{
+        ul_obj += "<li>"+item+"</li>";
+    });
+    ul_obj += "</ul>";
+    $(obj).parents('.form-group').prepend(ul_obj);
+    return ul_obj;
+}
+function load_errors(error_list, obj){
```

Comment:
This function is setup twice, one after the other, for no reason.

Reply:
Sorry, just copy paste garbage